packages/engine/src/avatar/functions/createAvatar.ts

```diff
@@ -37,8 +50,23 @@ export const defaultAvatarHalfHeight = defaultAvatarHeight / 2
 export const createAvatar = (spawnAction: typeof WorldNetworkAction.spawnAvatar.matches._TYPE): Entity => {
   const world = Engine.instance.currentWorld
   const userId = spawnAction.$from
+  const existingAvatarEntity = world.getUserAvatarEntity(spawnAction.$from)
   const entity = world.getNetworkObject(spawnAction.$from, spawnAction.networkId)!
 
+  // already spawned into the world on another device or tab
+  if (existingAvatarEntity) {
+    const didSpawnEarlierThanThisClient = NetworkPeerFunctions.getCachedActionsForUser(userId).find(
+      (action) =>
+        WorldNetworkAction.spawnAvatar.matches.test(action) &&
+        action !== spawnAction &&
+        action.$time > spawnAction.$time
+    )
+    if (didSpawnEarlierThanThisClient) {
+      hasComponent(entity, NetworkObjectAuthorityTag) && removeComponent(entity, NetworkObjectAuthorityTag)
+    }
+    return UndefinedEntity
```

Comment:
shouldn't we return the existing entity?

Reply:
We should probably rename this function to `spawnAvatarReceptor`, for clarity, and to avoid confusing with `AvatarService.createAvatar()`

Reply:
> On the other hand, this function is functionally just an action receptor (for the spawnAvatar action), and it's odd for it to be returning anything at all.

we are using it in tests as the entities are hidden in the receiveSpawnObject receptor, we can easily get around this